src/Float.mo

```diff
@@ -471,47 +509,63 @@ module {
   /// ```
   public let fromInt : Int -> Float = Prim.intToFloat;
 
-  /// Returns `x == y`.
+  /// Determines whether `x` is equal to `y` within the defined tolerance of `epsilon`.
+  /// The `epsilon` considers numerical erros, see comment above.
+  /// Equivalent to `Float.abs(x - y) <= epsilon` for a non-negative epsilon.
   ///
-  /// Note: This operation is discouraged as it does not consider numerical errors, see comment above.
+  /// Traps if `epsilon` is negative or `nan`.
   ///
   /// Special cases:
   /// ```
-  /// equal(+0.0, -0.0) => true
-  /// equal(-0.0, +0.0) => true
-  /// equal(+inf, +inf) => true
-  /// equal(-inf, -inf) => true
-  /// equal(nan, nan)   => false
+  /// equal(+0.0, -0.0, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(-0.0, +0.0, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(+inf, +inf, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(-inf, -inf, epsilon) => true for any `epsilon >= 0.0`
+  /// equal(x, nan, epsilon)     => false for any x and `epsilon >= 0.0`
+  /// equal(nan, y, epsilon)     => false for any y and `epsilon >= 0.0`
   /// ```
   ///
   /// Example:
   /// ```motoko
   /// import Float "mo:base/Float";
   ///
-  /// Float.equal(-12.3, -1.23e1) // => true
+  /// let epsilon = 1e-6;
+  /// Float.equal(-12.3, -1.23e1, epsilon) // => true
   /// ```
-  public func equal(x : Float, y : Float) : Bool { x == y };
+  public func equal(x : Float, y : Float, epsilon : Float) : Bool {
+    if (not (epsilon >= 0.0)) {
+      // also considers nan, not identical to `epsilon < 0.0`
+      Prim.trap("epsilon must be greater or equal 0.0")
+    };
+    x == y or abs(x - y) <= epsilon // `x == y` to also consider infinity equal
+  };
 
-  /// Returns `x != y`.
+  /// Determines whether `x` is not equal to `y` within the defined tolerance of `epsilon`.
+  /// The `epsilon` considers numerical erros, see comment above.
+  /// Equivalent to `not equal(x, y, epsilon)`.
   ///
-  /// Note: This operation is discouraged as it does not consider numerical errors, see comment above.
+  /// Traps if `epsilon` is negative or `nan`.
```

Comment:
```suggestion
  /// Traps if `epsilon` is negative or `NaN`.
```

Reply:
I adjusted it for all occurrences in the documentation.